### PR TITLE
bug of catalog plots

### LINKED ISSUE #2299: No response except a blank screen when plot 2D-scatter based on loaded catalog
**Describe the bug**
Firstly, I opened an image and loaded the NVSS catalog. Then clicked the plot of image overlay based on DEC and RA. Finally, I want to plot 2D-scatter but the screen turns white and there is nothing showing in Carta, so I have to exit and do it again.

**To Reproduce**
Steps to reproduce the behavior (added by @kswang1029 ):
1. download https://drive.google.com/file/d/1xLw9MpklLnfCZqSDTvZmdBV7IbM29n9z/view?usp=drive_link
2. load the image and enable online catalog query dialog
3. switch to VizieR service and enter keyword "NVSS" to search 
4. select the first returned catalog and load it
Screenshot 2023-11-28 at 14 39 34
5. select RAJ2000 and DEJ2000 columns for catalog overlay rendering, this should work
6. switch to 2D scatter plot rendering using the same columns and the frontend will crash due to "formatTickValues"
Screenshot 2023-11-28 at 14 34 54

**Expected behavior**
I want to use 2D-scatter plot normally.

**Screenshots or videos**
If applicable, add screenshots/videos to help explain your problem.

https://github.com/CARTAvis/carta/assets/81305215/5318a3f8-5797-4512-ae6f-07cc00e90d3e

**Platform info (please complete the following information):**
 - OS: [e.g. iOS] Big Sur 11.5.2
 - Browser [e.g. chrome, safari] local application
 - Version [e.g. 22] 4.0.0

**Additional context**
It is the same situation when I use the histogram plot.

### PR DESCRIPTION
**Pull request by loveluthien**
**Description**

Fix #2299. Somehow the axis ranges from VizierR downloaded data will be the string type and makes the crash.

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] changelog updated / no changelog update needed
- [x] unit test added (for functions with no dependenies)
- [x] API documentation added (for public variables and methods in stores)

For dependencies:
- [x] e2e test passing / ~corresponding fix added~ / ~new e2e test created~ New test will be added in a later stage shortly
- [x] ~protobuf version bumped~ / no protobuf version bumped needed
- [x] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [x] ~corresponding ICD test fix added (`BackendService` changed)~ / no ICD test fix needed (`BackendService` unchanged)
- [x] user manual prepared (for large new features)

### CODE REVIEW

**Review by kswang1029 — APPROVED** [review]
tested and looks good. no other regression found from e2e tests.

**Review by crocka — APPROVED** [review]

**Comment by loveluthien** [comment]
@YuHsuan-Hwang It is pretty weird. If I set ```let border: Border;``` and ```this.formatTickValues([border.xMin, border.xMax]);```, inside this.formatTickValues, the type of ```range[0]``` is still a string even if ```border.xMin``` is a number.

File: src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx [hunk 1]
```diff
@@ -604,8 +604,8 @@ export class CatalogPlotComponent extends React.Component<WidgetProps> {
         const difference = range[1] - range[0];
```
Comment (crocka) [inline src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx]: should we do Number(range[1]) - Number(range[0]) as well?

**Comment by YuHsuan-Hwang** [comment]
Looks like we are allowing string types [here](https://github.com/CARTAvis/carta-frontend/blob/dev/src/models/AbstractCatalogProfile/AbstractCatalogProfile.ts#L161), and perhaps we need to adjust how we process the loaded catalog [here](https://github.com/CARTAvis/carta-frontend/blob/dev/src/utilities/CatalogApiProcessed/CatalogApiProcessed.ts#L168).

If converting from strings to numbers when loading the catalog has performance impact, we could consider add documentations and convert it when creating plots.

File: src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx (hunk 1 above)
Comment (loveluthien) [reply src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx]: The difference between two "number" strings will be a number in JS. But we should convert the string to a number earlier.

**Comment by loveluthien** [comment]
@YuHsuan-Hwang @kswang1029 I now convert the VizieR string data to "number" right after querying. It seems that the performance doesn't change much with some simple tests.

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Fixed bug that the frontend crashes when deleting annotations if the export window is opened ([#2278](https://github.com/CARTAvis/carta-frontend/issues/2278)).
 * Fixed the performance issue when panning images ([#2291](https://github.com/CARTAvis/carta-frontend/issues/2291)).
 * Improved the dragging performance of 'Export Region' widget when the list of region/annotation to be exported is large. ([#1867](https://github.com/CARTAvis/carta-frontend/issues/1867)).
+* Fixed the crash when plotting some online query catalog data. ([#2321](https://github.com/CARTAvis/carta-frontend/pull/2321))
```
Comment (YuHsuan-Hwang) [inline CHANGELOG.md]: I suggest we revise it to "plotting online Vizier catalog data."

File: src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx [hunk 3]
```diff
@@ -880,11 +880,11 @@ export class CatalogPlotComponent extends React.Component<WidgetProps> {
             layout.xaxis.range = [border.xMin, border.xMax];
             layout.yaxis.range = [border.yMin, border.yMax];
             layout.yaxis.title = widgetStore.yColumnName;
-            layout.yaxis.tickformat = this.formatTickValues(layout.yaxis.range);
             layout.xaxis.tickformat = this.formatTickValues(layout.xaxis.range);
```
Comment (YuHsuan-Hwang) [inline src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx]: I prefer we use `[border.xMin, border.xMax]`.

**Review by YuHsuan-Hwang — COMMENTED** [review]
Overall the fix looks good. 👍

File: CHANGELOG.md (hunk 2 above)
Comment (loveluthien) [reply CHANGELOG.md]: okay.

File: src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx (hunk 3 above)
Comment (loveluthien) [reply src/components/CatalogOverlay/CatalogPlotComponent/CatalogPlotComponent.tsx]: I modified them.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Review by kswang1029 — APPROVED** [review]
Looks good. I will add some tests at a later stage shortly (need some time to implement).

**Comment by kswang1029** [comment]
e2e tests added for both SIMBAD and VizieR.

**Pull request merged**